tests/unittest_observer/FaceliftObserverTest.cpp

```diff
@@ -0,0 +1,76 @@
+#include <gtest/gtest.h>
+#include "IPCProxyBase.h"
+#include "InterfaceBase.h"
+#include <QSignalSpy>
+
+namespace {
+
+using namespace facelift;
+
+class Counter
+{
+public:
+    Counter() { m_value = 0; }
+
+    void incValue() {
+        m_value++;
+    }
+    int getValue() const {
+       return m_value;
+    }
+private:
+    int m_value;
+};
+
+class IPCProxyBaseTest : public ::testing::Test
+{
+public:
+    Counter c1;
+    Counter c2;
+    StandartObserver  < std::function<void()> > * obs1 = new StandartObserver  < std::function<void()> > (std::bind(&Counter::incValue, &c1) );
+    SingleTimeObserver< std::function<void()> > * obs2 = new SingleTimeObserver< std::function<void()> > (std::bind(&Counter::incValue, &c2) );
+
+    IPCProxyBase<InterfaceBase> proxyBase{nullptr};
+
+    ~IPCProxyBaseTest() {
+        delete obs1;
+        obs1 = nullptr;
+        delete obs2;
+        obs2 = nullptr;
+    }
+};
+
+TEST_F(IPCProxyBaseTest, testObservers)
+{
+    // Set observers to proxy
+    const auto expected = QVector<IObserver*>{obs1, obs2,};
+    proxyBase.m_readyObserver.setObservers(expected);
+    const auto actual = proxyBase.m_readyObserver.getObservers();
+
+    // Check installed observers
+    EXPECT_EQ(expected.size(), actual.size());
+    EXPECT_EQ(expected[0], actual[0]);
+    EXPECT_EQ(expected[1], actual[1]);
```

Comment:
I don't get this part. Are you testing setter/getter pair?

Reply:
This is just additional checking.

Reply:
Done